src/TaskModel.vala

```diff
@@ -239,6 +239,38 @@ public class Tasks.TaskModel : Object {
 
                 registry.refresh_backend_sync (collection_source.uid, null);
                 break;
+            case "google":
+                var collection_source = registry.find_extension (task_list, E.SOURCE_EXTENSION_COLLECTION);
+                var authorizer = (GData.Authorizer) new E.GDataOAuth2Authorizer (collection_source, typeof (GData.TasksService));
+                var service = new GData.TasksService (authorizer);
+                var uri = "https://www.googleapis.com/tasks/v1/users/@me/lists/%s";
+                var id = ((E.SourceResource) task_list.get_extension (
+                    E.SOURCE_EXTENSION_RESOURCE
+                )).identity.replace ("gtasks::", "");
+
+                GData.TasksTasklist tasklist = null;
+                try {
+                    tasklist = (GData.TasksTasklist) yield service.query_single_entry_async (
+                        GData.TasksService.get_primary_authorization_domain (),
+                        uri.printf (id),
+                        null,
+                        typeof (GData.TasksTasklist),
+                        null
+                    );
+                } catch (Error e) {
+                    if (e is GData.ServiceError.NOT_FOUND) {
+                        throw new Tasks.TaskModelError.BACKEND_ERROR (
+                            "Task list '%s' is no longer available in Google backend.",
+                            id
+                        );
+                    } else {
+                        throw e;
+                    }
+                }
```

Comment:
Don't know about the exact details of the error message, but I guess we are fine with whatever error is thrown here. Handling just this one particular error case differently seems to complicate things a bit too much.

In case this is needed, we can still re-add in the future.